crates/server/src/admin/handlers/blob.rs

```diff
@@ -161,28 +285,252 @@ pub async fn info_handler(
+                    "Successfully deleted metadata for blob {} (file remains on disk)",
+                    current_blob_id
+                );
+            }
+            Err(err) => {
+                tracing::warn!(
+                    "Failed to delete metadata for blob {}: {}",
+                    current_blob_id,
+                    err
+                );
+            }
+        }
+    }
+
+    if deleted_count > 0 {
+        tracing::info!(
+            "Successfully deleted metadata for {} blob(s) including chunks (files remain on disk)",
+            deleted_count
+        );
+        ApiResponse {
+            payload: BlobDeleteResponse {
                 blob_id,
-                size: blob_data.len() as u64,
+                deleted: true,
             },
         }
-        .into_response(),
+        .into_response()
+    } else {
+        ApiError {
+            status_code: StatusCode::INTERNAL_SERVER_ERROR,
+            message: "Failed to delete any blob metadata".to_owned(),
+        }
+        .into_response()
+    }
+}
+
+#[derive(Debug, Serialize, Copy, Clone)]
+pub struct BlobDeleteResponse {
+    blob_id: BlobId,
+    deleted: bool,
+}
+
+/// Get blob metadata via HEAD request
+///
+/// Returns blob metadata in HTTP headers without the actual blob content.
+/// This is efficient for checking blob existence and getting size info.
+/// Also detects and returns MIME type based on file content.
+pub async fn info_handler(
+    Path(blob_id): Path<String>,
+    Extension(state): Extension<Arc<AdminState>>,
+) -> impl IntoResponse {
+    let blob_id: BlobId = match blob_id.parse() {
+        Ok(id) => id,
+        Err(_) => {
+            return ApiError {
+                status_code: StatusCode::BAD_REQUEST,
+                message: "Invalid blob ID format".to_owned(),
+            }
+            .into_response();
+        }
+    };
+
+    let handle = state.store.clone().handle();
+    let blob_key = BlobMetaKey::new(blob_id);
+
+    match handle.get(&blob_key) {
+        Ok(Some(blob_meta)) => {
+            let content_type = detect_mime_type(&state, &blob_id)
+                .await
+                .unwrap_or_else(|| "application/octet-stream".to_string());
+            let etag = format!("\"{}\"", hex::encode(blob_meta.hash));
+
+            Response::builder()
+                .status(StatusCode::OK)
+                .header("Content-Length", blob_meta.size.to_string())
+                .header("Content-Type", &content_type)
+                .header("ETag", &etag)
+                .header("Cache-Control", "public, max-age=3600") // 1 hour cache
+                .header("X-Blob-ID", blob_id.to_string())
+                .header("X-Blob-Hash", hex::encode(blob_meta.hash))
+                .header("X-Blob-MIME-Type", &content_type)
+                .header(
+                    "Access-Control-Expose-Headers",
+                    "X-Blob-ID, X-Blob-Hash, X-Blob-MIME-Type, ETag",
+                )
```

Comment:
the only observable difference between `GET /blobs/{id}` and `HEAD /blobs/{id}` is the body is not available in a `HEAD` request

which means all these headers should be present in the `GET` response as well